runtime/vm/visible.c

```diff
@@ -158,12 +158,16 @@ checkVisibility(J9VMThread *currentThread, J9Class* sourceClass, J9Class* destCl
 				 */
 				if (NULL == sourceClass->nestHost) {
 					result = loadAndVerifyNestHost(currentThread, sourceClass, lookupOptions);
+					sourceClass = J9_CURRENT_CLASS(sourceClass);
```

Comment:
Small nitpick - these only need to be reloaded if the visibility succeeds so this could move after the `if`